comment/models/reactions.py

```diff
@@ -0,0 +1,90 @@
+from collections import namedtuple
+from typing import List
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.db import models
+from django.utils import timezone
+from django.utils.translation import gettext_lazy as _
+
+from comment.manager.reactions import ReactionManager
+from comment.models.comments import Comment
+
+REACTION_CHOICES = [
+    (1, _('like')),
+    (2, _('dislike'))
+]
+
+# construct a named tuple
+choices = namedtuple('reaction_choices', ['value', 'reaction'])
+
+# construct the list of named tuple
+reaction_choices = [choices(*reaction) for reaction in REACTION_CHOICES]
+
+
+class Reaction(models.Model):
+    reactions = models.SmallIntegerField(choices=REACTION_CHOICES, default=reaction_choices[0].value)
```

Comment:
In this model design, a new row in the db will be created for each new reaction. More efficient solution could be as follows:
``` python
from django.db import models

class Reaction(models.Model):
    comment = models.ForeignKey(Comment, related_name='comment_reaction', on_delete=models.CASCADE)
    like = models.PositiveIntegerField(default=0)
    dislike = models.PositiveIntegerField(default=0)
    
    objects = models.Manager()
    comment_objects = ReactionManager()

    def increase_like(self):
        self.like = models.F('like') + 1
        self.save()
        
    def increase_dislike(self):
        self.dislike = models.F('dislike') + 1
        self.save()

    def decrease_like(self):
        if self.like > 0:
            self.like = models.F('like') - 1
            self.save()
            
    def decrease_dislike(self):
        if self.dislike > 0:
            self.dislike = models.F('dislike') - 1
            self.save()
      
```

I believe we could get better performance on this model since only one reaction entry will be created for each comment.

Reply:
Hey @Radi85 , by this design we don't associate a user to a reaction. This will raise a couple of issues:
- One user can like/dislike a comment multiple times.
 - We wouldn't be able to extend this (maintaining backward compatibilities) to show users if we want it in the future.

Reply:
Hey,
The user has nothing to do with this model because the user is still connected with the other mode `ReactionInstance`.
to keep traking of the user rection type we could add one more field `reaction_type` to `ReactionInstance` model.
The user will be able at anytime to like, dislike or remove his action.

Reply:
Okay, I misunderstood it. I thought you were asking to remove the `ReactionInstance` model. So I had used the reactions field so that we could extend the model in the future if we want more reactions like love, confused, etc. So, do you still want me to go through with your suggestion?

Reply:
Yes please. Adding new columns for new reactions is much more cheaper than creating new entry in the db for each new reaction.